pyproject.toml

```diff
@@ -131,7 +125,21 @@ enabler = [
 ]
 
 type = [
+	# upstream
 	"pytest-mypy",
+
+	# local
+
+	# pin mypy version so a new version doesn't suddenly cause the CI to fail,
+	# until types-setuptools is removed from typeshed.
+	# For help with static-typing issues, or mypy update, ping @Avasam 
+	"mypy==1.11.*",
+	# Conditionally imported vendored dependencies.
+	# They're not needed at runtime for Python version outside the [core] extra,
+	# but they're still needed for type-checking since we import them directly
+	"tomli",
+	"importlib_resources",
+	"importlib_metadata",
```

Comment:
I wonder, can we have these conditional on the Python version?

I think it would be interesting from the testing perspective to see how the test suite behaves when these packages are not installed.

In the `_importlib.py` file we have `if sys.version_info ...` protecting the imports, do you think the typechecking would be smart enough to deal with that?

Reply:
Yes, type checkers should understand simple version and platform checks.

Reply:
That is an interesting point.

What happens if we run `mypy` (via `pytest-mypy`) with the Python 3.12 executable, but in `mypy.ini` we have `python_version = 3.8`? Will `mypy` simulate that the file is running under `3.8` even for things like `sys.version_info`?

Reply:
Exactly. And it'll take into account modules that are present (or not) in stdlib as specified by typeshed. You could run all your version and platform checks from the same Python executable (as long as all your packages are present in your environment and you don't use new syntax that required AST changes)

Reply:
Thank you for the clarification @Avasam. 

Now that #4352 is merged should we remove `tomli`, `importlib_*` so that the test suite can check against the stdlib's version? Or are these dependencies still needed?

Reply:
We should be able to remove them. I'm on it :)